components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultURLResolverService.java

```diff
@@ -0,0 +1,201 @@
+ */
+public class DefaultURLResolverService implements URLResolverService {
+
+    @Override
+    public String resolveUrl(String url, boolean addProxyContextPath, boolean addWebContextRoot,
+                             Map<String, Object> properties) throws URLResolverException {
+
+        try {
+            URL parsedUrl = new URL(url);
+            StringBuilder urlBuilder = new StringBuilder(parsedUrl.getProtocol())
+                    .append("://")
+                    .append(parsedUrl.getHost())
+                    .append(":")
+                    .append(parsedUrl.getPort());
+            appendContextToUri(parsedUrl.getPath(), addProxyContextPath, addWebContextRoot, urlBuilder,
+                    false);
+            return urlBuilder.toString();
+
+        } catch (MalformedURLException e) {
+            throw new URLResolverException("Error while parsing the URL: " + url, e);
+        }
+    }
+
+    @Override
+    public String resolveUrlContext(String urlContext, boolean addProxyContextPath, boolean addWebContextRoot,
+                                    boolean addTenantQueryParamInLegacyMode, Map<String, Object> properties)
+            throws URLResolverException {
+
+        String hostName = getHostName();
+        String mgtTransport = CarbonUtils.getManagementTransport();
+        int mgtTransportPort = getMgtTransportPort(mgtTransport);
+
+        if (hostName.endsWith("/")) {
+            hostName = hostName.substring(0, hostName.length() - 1);
+        }
+        StringBuilder serverUrl = new StringBuilder(mgtTransport).append("://").append(hostName.toLowerCase());
+        // If it's well known HTTPS port, skip adding port.
+        if (mgtTransportPort != IdentityCoreConstants.DEFAULT_HTTPS_PORT) {
+            serverUrl.append(":").append(mgtTransportPort);
+        }
+
+        appendContextToUri(urlContext, addProxyContextPath, addWebContextRoot, serverUrl, addTenantQueryParamInLegacyMode);
+
+        return serverUrl.toString();
+    }
+
+    private int getMgtTransportPort(String mgtTransport) {
+
+        AxisConfiguration axisConfiguration = IdentityCoreServiceComponent.getConfigurationContextService().
+                getServerConfigContext().getAxisConfiguration();
+        int mgtTransportPort = CarbonUtils.getTransportProxyPort(axisConfiguration, mgtTransport);
+        if (mgtTransportPort <= 0) {
+            mgtTransportPort = CarbonUtils.getTransportPort(axisConfiguration, mgtTransport);
+        }
+        return mgtTransportPort;
+    }
+
+    private String getHostName() throws URLResolverException {
+
+        String hostName = ServerConfiguration.getInstance().getFirstProperty(IdentityCoreConstants.HOST_NAME);
+        try {
+            if (hostName == null) {
+                hostName = NetworkUtils.getLocalHostname();
+            }
+        } catch (SocketException e) {
+            throw new URLResolverException("Error while trying to read hostname.", e);
```

Comment:
Is it reading hostname. This should denote that trying to resolve hostname from system